ios-swiftui/realm-template-apps-swiftui/Views/LogoutButton.swift

```diff
@@ -4,19 +4,54 @@ import RealmSwift
 /// Logout from the synchronized realm. Returns the user to the login/sign up screen.
 struct LogoutButton: View {
     @State var isLoggingOut = false
+    @State var error: Error?
+    @State var errorMessage: ErrorMessage? = nil
     
     var body: some View {
+        if isLoggingOut {
+            ProgressView()
+        }
         Button("Log Out") {
             guard let user = realmApp.currentUser else {
                 return
             }
             isLoggingOut = true
-            user.logOut() { error in
-                isLoggingOut = false
+            Task {
+                await logout(user: user)
                 // Other views are observing the app and will detect
                 // that the currentUser has changed. Nothing more to do here.
-                print("Logged out")
+                isLoggingOut = false
             }
         }.disabled(realmApp.currentUser == nil || isLoggingOut)
+        // Show an alert if there is an error during logout
+        .alert(item: $errorMessage) { errorMessage in
+            Alert(
+                title: Text("Failed to log out"),
+                message: Text(errorMessage.errorText),
+                dismissButton: .cancel()
+            )
+        }
+    }
+    
+    /// Asynchronously log the user out, or display an alert with an error if logout fails.
+    func logout(user: User) async {
+        do {
+            try await user.logOut()
+            print("Successfully logged user out")
+        } catch {
+            print("Failed to log user out: \(error.localizedDescription)")
+            // Realm returns an error when logout fails. Capture the error.
+            self.error = error
+            // SwiftUI Alert requires the item it displays to be Identifiable.
+            // Optional Error is not Identifiable.
+            // Store the error as errorText in our Identifiable ErrorMessage struct,
+            // which we can display in the alert.
+            self.errorMessage = ErrorMessage(errorText: error.localizedDescription)
```

Comment:
In that case, why not do the reverse and store the original error object in the ErrorMessage object? This reduces the amount of unnecessary state held onto by the LogoutButton view. (In other words, you only have to nil out one property instead of two to go back to "not error" state)